integration_tests/src/main/python/csv_test.py

```diff
@@ -622,9 +623,9 @@ def do_read(spark):
             non_exist_classes = cpu_scan_class,
             conf = conf)
 
-@allow_non_gpu('FileSourceScanExec', 'CollectLimitExec', 'DeserializeToObjectExec', *non_utc_allow)
+@allow_non_gpu('FileSourceScanExec', 'CollectLimitExec', 'DeserializeToObjectExec')
 @pytest.mark.skipif(is_before_spark_340(), reason='`preferDate` is only supported in Spark 340+')
-
+@pytest.mark.xfail(is_not_utc(), reason='Timezone is not supported for csv format as https://github.com/NVIDIA/spark-rapids/issues/9653.')
```

Comment:
So then how does https://github.com/NVIDIA/spark-rapids/issues/9653 cover that? I don't see CSV or JSON mentioned in there at all. I don't wee any mention of timestamp_ntz being mentioned. I just want to be 100% sure that we don't end up dropping something and getting data corruption that we missed because of an xfail. I would rather see a complex test, or at a minimum an issue filed to do the test correctly.

Reply:
Updated by refining the test cases.